pkgs/leak_tracker/lib/src/leak_tracking/_object_tracker.dart

```diff
@@ -98,6 +102,18 @@ class ObjectTracker implements LeakProvider {
     _objects.assertRecordIntegrity(record);
   }
 
+  /// Declares all not disposed objects as leaks, even if they are not GCed yet.
+  ///
+  /// Is used to make sure all disposables are disposed by the the end of the test.
+  void declareAllNotDisposedAsLeaks() {
+    throwIfDisposed();
+    final notGCedAndNotDisposed = <ObjectRecord>[];
+    _objects.notGCed.forEach((record) {
```

Comment:
Nit: lots of projects have lints recommending that we don't use `forEach`. Even if we don't have it enabled for this project, consider doing this instead:

```dart
for (final record in _objects.notGCed) {
    if (!record.isDisposed) {
      _declareNotDisposedLeak(record);
    }
}
```

Reply:
The lint is enabled for this project. notGCed is not iterable though. I will check if it is easy to make it iterable.

Reply:
Oh interesting. At the very least, I don't think you need `noGCedAndNotDisposed`, but you could instead do:

```dart
_objects.notGCed.forEach((record) {
  if (!record.isDisposed) {
    _declareNotDisposedLeak(record);
  }
});
```

Reply:
Yes, looks as optimizable. Added comment why I did it:

```
// We need this temporary storage to avoid errror 'concurrent modification during iteration'
// for internal iterables in `_objects.notGCed`.
```

Reply:
And I checked: Iterable wants 27 methods to be implemented. Seems to be too heavy.